test/cppsuite/tests/search_near_01.cxx

```diff
@@ -0,0 +1,256 @@
+                for (uint64_t k = 0; k < ALPHABET_SIZE; ++k) {
+                    for (uint64_t count = 0; count < tc->key_count; ++count) {
+                        tc->transaction.begin();
+                        /*
+                         * Generate the prefix key, and append a random generated key string based
+                         * on the key size configuration.
+                         */
+                        prefix_key = {alphabet.at(tc->id), alphabet.at(j), alphabet.at(k)};
+                        prefix_key += random_generator::instance().generate_random_string(
+                          tc->key_size - PREFIX_KEY_LEN);
+                        if (tc->insert(cursor, coll.id, prefix_key)) {
+                            testutil_assert(rollback_retries < MAX_ROLLBACKS);
+                            /* We failed to insert, rollback our transaction and retry. */
+                            tc->transaction.rollback();
+                            rollback_retries++;
+                            --count;
+                            continue;
+                        }
+                        /* Commit txn at commit timestamp 100. */
+                        tc->transaction.commit("commit_timestamp=" + tc->tsm->decimal_to_hex(100));
+                        rollback_retries = 0;
+                    }
+                }
+            }
+        }
+    }
+
+    public:
+    search_near_01(const test_harness::test_args &args) : test(args) {}
+
+    void
+    populate(test_harness::database &database, test_harness::timestamp_manager *tsm,
+      test_harness::configuration *config, test_harness::workload_tracking *tracking) override final
+    {
+        uint64_t collection_count, key_count, key_size;
+        std::vector<thread_context *> workers;
+        thread_manager tm;
+
+        /* Validate our config. */
+        collection_count = config->get_int(COLLECTION_COUNT);
+        key_count = keys_per_prefix = config->get_int(KEY_COUNT_PER_COLLECTION);
+        key_size = config->get_int(KEY_SIZE);
+        testutil_assert(key_size >= PREFIX_KEY_LEN);
+        /* Keys must be unique. */
+        testutil_assert(key_count <= pow(10, key_size));
+
+        logger::log_msg(LOG_INFO,
+          "Populate configuration with key size: " + std::to_string(key_size) +
+            " key count: " + std::to_string(key_count) +
+            " number of collections: " + std::to_string(collection_count));
+
+        /* Create n collections as per the configuration. */
+        for (uint64_t i = 0; i < collection_count; ++i)
+            /*
+             * The database model will call into the API and create the collection, with its own
+             * session.
+             */
+            database.add_collection();
+
+        /*
+         * Spawn thread_count threads to populate the database, theoretically we should be IO bound
+         * here.
+         */
+        for (uint64_t i = 0; i < ALPHABET_SIZE; ++i) {
```

Comment:
Could you follow the example given [here](https://github.com/wiredtiger/wiredtiger/blob/develop/test/cppsuite/test_harness/workload/database_operation.cxx#L103-L112)? Is there a reason why the `std::bind` approach is required/better? If it is, we could change the example in `database_operation.cxx`.

Reply:
If you follow the example, you don't need to define `tsm` and `tracking`. You can just remove their names from the function arguments but still leave the types.

Reply:
the original reason behind `std::bind`, allowed me to use the constant variables alphabet and ALPHABET_SIZE within the class. Let me know what you think of this approach.

Reply:
Thanks! much cleaner now.